python/iceberg/api/expressions/inclusive_manifest_evaluator.py

```diff
@@ -159,3 +159,7 @@ def in_(self, ref, lit):
 
     def not_in(self, ref, lit):
         return ROWS_MIGHT_MATCH
+
+    def starts_with(self, ref, lit):
+        # if it's all null then we can't match :)
+        return self.not_null(ref)
```

Comment:
Most of the functions seem to explicitly return the constants `ROWS_MIGHT_MATCH` and `ROWS_CANNOT_MATCH`. Would this fit in more as `return ROWS_CANNOT_MATCH if self.not_null(ref) else ROWS_MIGHT_MATCH` or similar?

Reply:
I don't have a strong opinion here but the convention in the java library is that the leaf predicates always return either of those constants.

Reply:
I don't have strong views here either, my only consideration with changing it to `return ROWS_CANNOT_MATCH if self.not_null(ref) else ROWS_MIGHT_MATCH` would be if we ever change this from wrapping ROWS_CANNOT_MATCH/ROWS_MIGHT_MATCH true/false it would brake but we'd catch it.